client/src/main/java/org/eclipse/hono/client/AsyncCommandClient.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client;
+
+import java.util.Map;
+
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A client for sending asynchronous request response commands, i.e. all command responses for a specific tenant and
+ * reply ID shall be received by the same consumer in an asynchronous fashion.
```

Comment:
IMHO this is what we do not want to enforce by introducing the `AsyncCommandResponseConsumer`, isn't it?

Reply:
I don't get your point here.

Reply:
Well, my point is that we specifically introduced the `AsyncCommandResponseConsumer` so that multiple instances of it can consume responses for arbitrary commands sent using the same replyId. So the intention is *not* that
> all command responses for a specific tenant and reply ID shall be received by the **same** consumer

Reply:
I see, I was referring to the same consumer from a logical point of view and not the very same instance, i.e. using the same replyId. I must admit wording is wording and confusing. I will change.

Reply:
I just deleted the i.e. part